api/envoy/extensions/filters/http/jwt_authn/v3/config.proto

```diff
@@ -272,6 +272,14 @@ message JwtProvider {
   //
   string header_in_metadata = 14;
 
+  // If true, clear the route cache in order to allow this filter to impact routing decisions.
+  // When a JWT provider executes header or metadata operations, e.g. via
+  // :ref:`payload_in_metadata <envoy_v3_api_field_extensions.filters.http.jwt_authn.v3.JwtProvider.payload_in_metadata>`
+  // the route should be re-resolved before forwarding to upstream.
+  // If not specified, default is false.
+  // [#not-implemented-hide:]
+  bool clear_route_cache = 15;
```

Comment:
Thank for adding this! (I planned to add this but somehow it gets lost in my backlog)